tests/testmongodb.nim

```diff
@@ -0,0 +1,219 @@
+import unittest
+
+import os, strutils, sequtils, times
+
+import norm / mongodb
+
+
+const
+  # for local testing, modify your /etc/hosts file to contain "mongodb_1"
+  # pointing to your local mongodb server
+  dbConnection = "mongodb://mongodb_1:27017"
+  dbName = "TestDb"
+  customDbName = "TestCustomDb"
+
+type
+  Copyright = object
+    legal_name: string
+    year: int
+    # year {.dbCol: "yr".}: int
+
+dbAddObject(Copyright)
+
+db(dbConnection, "", "", dbName):
+  type
+    User {.table: "users".} = object
+      email {.unique.}: string
+      ssn: Option[int]
+      birthDate: Time
+    Publisher {.table: "publishers".} = object
+      title {.unique.}: string
+    Book {.table: "books".} = object
+      title: string
+      authorEmail {.fk: User.email, onDelete: "CASCADE".}: string
+      publisherTitle {.fk: Publisher.title.}: string
+      ratings: seq[float]
+      copyright: Copyright
+
+  # TODO proc getBookById(id: Oid): Book = withDb(Book.getOne id)
+  # TODO add and test foreign key pulls in mongo
+
+type

```

Comment:
Looks like a typo. This should be indented under `db` scope.

Reply:
Or maybe it isn't. I see you've added `dbAddCollection` which apparently allows to add table definitions separately.

Reply:
One of the benefits is that objects from other libraries can also be stored now; whereas before you could only add objects you had control over.